frontend/src/utils/enginePower.ts

```diff
@@ -0,0 +1,190 @@
+  endSuperchargerendAltitude?: number,
+  turbocharger?: number
+) => {
+  let altitudes = Array.from(
+    { length: Math.floor(1000*maxAltitude / 100) },
+    (_, i) =>  100*i
+  );
+  altitudes = [...altitudes, 1000*maxAltitude];
+
+  if (
+    halfSuperchargerendAltitude &&
+    altitudes.includes(halfSuperchargerendAltitude) === false
+  ) {
+    altitudes = [...altitudes, halfSuperchargerendAltitude];
+    altitudes.sort(function (a, b) {
+      return a - b;
+    });
+  }
+
+  if (
+    endSuperchargerendAltitude &&
+    altitudes.includes(endSuperchargerendAltitude) === false
+  ) {
+    altitudes = [...altitudes, endSuperchargerendAltitude];
+    altitudes.sort(function (a, b) {
+      return a - b;
+    });
+  }
+
+  if (turbocharger && altitudes.includes(turbocharger) === false) {
+    altitudes = [...altitudes, turbocharger];
+    altitudes.sort(function (a, b) {
+      return a - b;
+    });
+  }
+  return altitudes;
+};
+
+const powerArrayCreator = (altitude: number, props: powerInput) => {
+  let power = normalPower(altitude, 0, props.seaLevelPower, props.kCoefficient);
+
+  if (props.engineType === "piston") {
+    if (
+      props.halfSupercharger &&
+      altitude <= props.halfSupercharger.endAltitude
+    ) {
+      power = superchargerPower(
+        altitude,
+        0,
+        props.halfSupercharger.endAltitude,
+        props.seaLevelPower,
+        props.halfSupercharger.endPower
+      );
+    } else if (
+      props.halfSupercharger &&
+      altitude > props.halfSupercharger.endAltitude
+    ) {
+      power = normalPower(
+        altitude,
+        props.halfSupercharger.endAltitude,
+        props.halfSupercharger.endPower,
+        props.kCoefficient
+      );
+    }
+
+    if (
+      props.halfSupercharger &&
+      props.endSupercharger &&
+      altitude >= props.endSupercharger.startAltitude &&
+      altitude <= props.endSupercharger.endAltitude
+    ) {
+      let startPower = normalPower(
+        props.endSupercharger.startAltitude,
+        props.halfSupercharger.endAltitude,
+        props.halfSupercharger.endPower,
+        props.kCoefficient
+      );
+      power = superchargerPower(
+        altitude,
+        props.endSupercharger.startAltitude,
+        props.endSupercharger.endAltitude,
+        startPower,
+        props.endSupercharger.endPower
+      );
+    } else if (
+      props.endSupercharger &&
+      altitude > props.endSupercharger.endAltitude
+    ) {
+      power = normalPower(
+        altitude,
+        props.endSupercharger.endAltitude,
+        props.endSupercharger.endPower,
+        props.kCoefficient
+      );
+    } else if (props.turbocharger)
+      power = powerTurbocharged(
+        props.turbocharger,
+        props.seaLevelPower,
+        altitude,
+        props.kCoefficient
+      );
+  } else if (props.engineType === "turbine") {
+    power = turbinePower(altitude, props.seaLevelPower);
+  }
+
+  return power;
+};
+
+function normalPower(
```

Comment:
You mix arrow functions and the ones created with  the`function` keyword. Use arrows only